src/main/java/org/opensearch/security/user/UserTokenHandler.java

```diff
@@ -0,0 +1,184 @@
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.util.Collections;
+
+public class UserTokenHandler {
+
+    private final int DEFAULT_EXPIRATION_TIME_SECONDS = 300;
+    JwtVendor jwtVendor;
+    Settings settings;
+
+    ClusterService clusterService;
+
+    ConfigurationRepository configurationRepository;
+
+    ThreadPool threadPool;
+
+    JwtVerifier jwtVerifier;
+
+    String claimsEncryptionKey = RandomStringUtils.randomAlphanumeric(16);
+
+    String signingKey = RandomStringUtils.randomAlphanumeric(16);
+
+    Client client;
+
+    final static String FAILED_CLEAR_HASH_MESSAGE = "The hash could not be cleared from the specified account.";
+
+    public static CType getRevokedTokensConfigName() {
+        return CType.REVOKEDTOKENS;
+    }
+
+    @Inject
+    public UserTokenHandler(
+        ThreadPool threadPool,
+        ClusterService clusterService,
+        ConfigurationRepository configurationRepository,
+        Client client
+    ) {
+        this.settings = Settings.builder().put("signing_key", signingKey).put("encryption_key", claimsEncryptionKey).build();
+        this.jwtVendor = new JwtVendor(settings, null);
+        this.threadPool = threadPool;
+        this.clusterService = clusterService;
+        this.client = client;
+        this.configurationRepository = configurationRepository;
+    }
+
+    // public AuthToken issueToken(String audience) {
+    // ThreadContext threadContext = threadPool.getThreadContext();
+    // final User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+    // String jwt = null;
+    // try {
+    // jwt = jwtVendor.createJwt(
+    // clusterService.getClusterName().toString(),
+    // user.getName(),
+    // audience,
+    // DEFAULT_EXPIRATION_TIME_SECONDS,
+    // new ArrayList<String>(user.getRoles()),
+    // null
+    // );
+    // } catch (Exception e) {
+    // throw new RuntimeException(e);
+    // }
+    // return new BearerAuthToken(jwt);
+    // }
+
+    public boolean validateToken(AuthToken authToken) {
+        if (!(authToken instanceof BearerAuthToken)) {
+            throw new UserServiceException("The provided token is not a BearerAuthToken.");
+        }
+        BearerAuthToken bearerAuthToken = (BearerAuthToken) authToken;
+        JwsJwtCompactConsumer jwtConsumer = new JwsJwtCompactConsumer(bearerAuthToken.getCompleteToken());
+        JwtToken jwt = jwtConsumer.getJwtToken();
+
+        Long iat = (Long) jwt.getClaim("iat");
+        Long exp = (Long) jwt.getClaim("exp");
+        SecurityDynamicConfiguration revokedTokens = load(getRevokedTokensConfigName(), false);
+        Long currentTime = System.currentTimeMillis();
+        return (exp > currentTime && !revokedTokens.exists(bearerAuthToken.getCompleteToken()));
+    }
+
+    public boolean validateJustInTimeToken(AuthToken authToken) {
```

Comment:
The reason I make this validation is that, I think we do have different expectations for obo tokens than the other bearer auth tokens. For obo tokens, according to its feature of just-in-time, we do expect the payload of `iat` and `nbf` have the same value.